### PR TITLE
Reduce list spacing

### LINKED ISSUE #4082: [Bug]: Reduce padding in lists inside of messages
### Is there an existing issue for the same bug?

- [X] I have checked the troubleshooting document at https://docs.all-hands.dev/modules/usage/troubleshooting
- [X] I have checked the existing issues.

### Describe the bug

The spacing here is a little over the top :) 
Screenshot 2024-09-27 at 10 39 35 AM

This can probably be fixed by modifying the spacing in `frontend/src/index.css`

### Current OpenHands version

```bash
0.9.5
```

### Installation and Configuration

```bash
`docker run` from the README
```

### Model and Agent

N/A

### Operating System

N/A

### Reproduction Steps

Write a prompt that uses markdown bullet points

### Logs, Errors, Screenshots, and Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Vaishakh-SM**
**Give a summary of what the PR does, explaining any non-trivial design decisions**

This PR makes two CSS improvements:

1. Moves list-related CSS from index.css to the Markdown component.
2. Reduces list item spacing by adjusting the line height.

**Before**
Screenshot from 2024-10-02 21-31-41

**After**
Screenshot from 2024-10-02 21-33-14

---
**Link of any specific issues this addresses**

Resolves #4082

### CODE REVIEW

**Comment by tobitege** [comment]
This already looks much better, thanks for working on this!

Though not in the original issue, there's still quite a gap above/below the list (not sure which padding that is), but would you mind making that a little narrower, too, please?

**Comment by Vaishakh-SM** [comment]
> Though not in the original issue, there's still quite a gap above/below the list (not sure which padding that is), but would you mind making that a little narrower, too, please?

Sure!

I also plan on addressing the lint errors related to this change.

**Comment by Vaishakh-SM** [comment]
I've made the following changes:

- Removed an unused import from the file (was giving a lint error)
- Moved ul and ol components to a new file under the markdown directory (similar to the code file)
- Reduced the spacing between markdown elements.

It seems like the spacing was not caused by padding or the margin of the list, but rather by the spacing between children of the markdown. (Please correct me if I'm wrong here). Reducing the spacing here seemed aesthetically more pleasing.

Screenshot from 2024-10-03 01-17-25

**Comment by tobitege** [comment]
Thanks! Our UI specialist will have a final look once he's got time, so please bear with us. 👍

File: frontend/src/components/chat/ChatMessage.tsx [hunk 1]
```diff
@@ -89,7 +89,14 @@ function ChatMessage({
           {isCopy ? <FaClipboardCheck /> : <FaClipboard />}
         </button>
       )}
-      <Markdown components={{ code }} remarkPlugins={[remarkGfm]}>
+      <Markdown
```
Comment (Vaishakh-SM) [inline frontend/src/components/chat/ChatMessage.tsx:92]: Oops, seems like I removed remarkGfm by mistake. Ill add this back in

**Comment by rbren** [comment]
Looking much better! I might also shrink the space before and after the list a bit

**Comment by Vaishakh-SM** [comment]
> Looking much better! I might also shrink the space before and after the list a bit

Cool!
I've reduced the spacing between before and after the list a bit more. From my understanding, this spacing is caused by the spacing between children of the Markdown component; I've reduced the spacing between the children to shrink the space between them. So this won't be specific to lists but to any children of the mardown component.

Please let me know if this is fine or if we need to follow a different approach here.

**Comment by tobitege** [comment]
> Please let me know if this is fine or if we need to follow a different approach here.

May I ask for an example screenie of the newest version, please? 😃

**Comment by Vaishakh-SM** [comment]
> May I ask for an example screenie of the newest version, please? 😃

Sure! 😃
Screenshot from 2024-10-04 00-41-07

**Comment by tobitege** [comment]
Thanks! I don't see a big difference above/below the list compared to the previous image, though? 🤔 
Sorry for being picky 😄

**Comment by Vaishakh-SM** [comment]
> Thanks! I don't see a big difference above/below the list compared to the previous image, though? 🤔 Sorry for being picky 😄

Does this look any better? 😄
Screenshot from 2024-10-04 00-59-18

**Comment by tobitege** [comment]
Yes! Personally, I like that last version best, awesome! Thanks!!
@rbren wdyt?

**Review by rbren — APPROVED** [review]
Great work!

**Pull request merged**